plugins/outputs/graphite/graphite.go

```diff
@@ -44,11 +70,28 @@ func (g *Graphite) Connect() error {
 	if len(g.Servers) == 0 {
 		g.Servers = append(g.Servers, "localhost:2003")
 	}
+
+	// Set tls config
+	var err error
+	g.tlsConfig, err = internal.GetTLSConfig(
+		g.SSLCert, g.SSLKey, g.SSLCA, g.InsecureSkipVerify)
+	if err != nil {
+		return err
+	}
+
 	// Get Connections
 	var conns []net.Conn
 	for _, server := range g.Servers {
-		conn, err := net.DialTimeout("tcp", server, time.Duration(g.Timeout)*time.Second)
+		var conn net.Conn
+		if g.SSLEnabled {
+			// If ssl is enabled make secure tcp connection
+			conn, err = tls.Dial("tcp", server, g.tlsConfig)
```

Comment:
Need to use timeout with tls connections.  Should be able to do this using tls.DialWithDialer.

I think its possible to just use a nil tls.Config to do a insecure connection, but not sure.  This way you don't need the branch.